modules/announcements/module.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Elementor\Modules\Announcements;
+
+use Elementor\Core\Base\App as BaseApp;
+use Elementor\Modules\Announcements\Classes\Announcement;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Module extends BaseApp {
+
+	/**
+	 * @return bool
+	 */
+	public static function is_active(): bool {
+		return is_admin();
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_name(): string {
+		return 'announcements';
+	}
+
+	/**
+	 * Render wrapper for the app to load
+	 */
+	private function render_app_wrapper() {
+		?>
+		<div id="e-announcements-root"></div>
+		<?php
+	}
+
+	/**
+	 * Enqueue app scripts
+	 */
+	private function enqueue_scripts() {
+		wp_enqueue_script(
+			'announcements-app',
+			$this->get_js_assets_url( 'announcements-app' ),
+			[],
+			ELEMENTOR_VERSION,
+			true
+		);
+
+		$this->print_config( 'announcements-app' );
+	}
+
+	/**
+	 * Get init settings to use in frontend
+	 * @return array[]
+	 */
+	protected function get_init_settings(): array {
+		$active_announcements = $this->get_active_announcements();
+		$additional_settings = [];
+
+		foreach ( $active_announcements as $announcement ) {
+			$additional_settings[] = $announcement->get_prepared_data();
+			//@TODO - replace with ajax request from the front after actually triggered
+			$announcement->after_triggered();
+		}
+
+		return [
+			'announcements' => $additional_settings,
+		];
+	}
+
+	/**
+	 * Enqueue the module styles.
+	 */
+	public function enqueue_styles() {
+		wp_enqueue_style(
+			'announcements-app',
+			$this->get_css_assets_url( 'modules/announcements/announcements' ),
+			[],
+			ELEMENTOR_VERSION
+		);
+	}
+
+	/**
+	 * Retrieve all announcement in raw format ( array )
+	 * @return array[]
+	 */
+	private function get_raw_announcements(): array {
+		//@TODO - Change to real data
```

Comment:
todo?

Reply:
Yes, it's phase 2

Reply:
this is now related to phase 2.
this is a reminder for you/us to replace the data to the correct one ( from the product )

Reply:
Wait, will this be a long long array in the file here, the more announcements are added?

Maybe already put them all in some separate file?

Reply:
In phase 2 we will get the data from the api.